plugins/xml/core/src/main/java/org/pentaho/di/ui/trans/steps/xmloutput/XMLOutputDialog.java

```diff
@@ -848,22 +851,17 @@ public void modifyText( ModifyEvent e ) {
       }
     } );
 
-    wbFilename.addSelectionListener( new SelectionAdapter() {
-      public void widgetSelected( SelectionEvent e ) {
-        FileDialog dialog = new FileDialog( shell, SWT.SAVE );
-        dialog.setFilterExtensions( new String[] { "*.xml", "*.txt", "*.csv", "*" } );
-        if ( wFilename.getText() != null ) {
-          dialog.setFileName( transMeta.environmentSubstitute( wFilename.getText() ) );
-        }
-        dialog.setFilterNames( new String[] { BaseMessages.getString( PKG, "System.FileType.XMLFiles" ),
-          BaseMessages.getString( PKG, "System.FileType.TextFiles" ),
-          BaseMessages.getString( PKG, "System.FileType.CSVFiles" ),
-          BaseMessages.getString( PKG, "System.FileType.AllFiles" ) } );
-        if ( dialog.open() != null ) {
-          wFilename.setText( dialog.getFilterPath() + System.getProperty( "file.separator" ) + dialog.getFileName() );
-        }
-      }
-    } );
+    wbFilename.addSelectionListener(
+            new SelectionAdapterFileDialogTextVar(
+                    log,
+                    wFilename,
+                    transMeta,
+                    new SelectionAdapterOptions(
+                            SelectionOperation.SAVE_TO,
+                            new FilterType[] { FilterType.XML, FilterType.TXT, FilterType.CSV, FilterType.ALL },
+                            FilterType.XML,
+                            new String[] {"local"},
```

Comment:
There's a constructor that will take ProviderFilterType Enum, check line 63 of SelectionAdapterOptions